### DISCUSSION TITLE
How to trace API methods

### DISCUSSION THREAD

**Discussion by kochia3** [yanivagman/BPFroid#4] [Q&A] [answered]
hooks.json file have api methods to hook, `getNetworkOperatorNamel`, `getLine1Number`, etc.
How is possible to uprobe such java methods and dump arguments?

How the project handles to trace such methods?

**Comment by yanivagman** [comment, marked as answer]
The method is explained in the paper: https://arxiv.org/abs/2105.14344
The idea is to take advantage of ART which compiles the Android API ahead of time, find the methods offsets, and use regular uprobes to attach them
